doc/index.rst

```diff
@@ -0,0 +1,268 @@
+
+4.1 Connections
+~~~~~~~~~~~~~~~
+Input:
+  * C++: void callback(const std::shared_ptr<M const>&)
+Output:
+  * C++: void callback(const std::shared_ptr<M const>&)
+
+4.2 Example (C++)
+~~~~~~~~~~~~~~~~~
+.. code-block:: C++
+
+     message_filters::Subscriber<example_interfaces::msg::String> sub(node, "my_topic", 1);
+     message_filters::TimeSequencer<example_interfaces::msg::String> seq(
+       sub, rclcpp::Duration(0.1), rclcpp::Duration(0.01), 10);
+     seq.registerCallback(myCallback);
+
+5. Cache
+--------
+Stores a time history of messages.
+
+Given a stream of messages, the most recent N messages are cached in a ring buffer, from which time intervals of the cache can then be retrieved by the client. The timestamp of a message is determined from its header field.
+
+If the message type doesn't contain a header, see below for workaround.
+
+The Cache immediately passes messages through to its output connections.
+
+5.1 Connections
+~~~~~~~~~~~~~~~
+Input:
+  * C++: void callback(const std::shared_ptr<M const>&)
+  * Python: callback(msg)
+Output:
+  * C++: void callback(const std::shared_ptr<M const>&)
+  * Python: callback(msg)
+
+5.2 Example (C++)
+~~~~~~~~~~~~~~~~~
+.. code-block:: C++
+
+    message_filters::Subscriber<example_interfaces::msg::String> sub(node, "my_topic", 1);
+    message_filters::Cache<example_interfaces::msg::String> cache(sub, 100);
+    cache.registerCallback(myCallback);
+
+5.3 Example (Python)
+~~~~~~~~~~~~~~~~~~~~
+.. code-block:: Python
+
+    sub = message_filters.Subscriber('my_topic', sensor_msgs.msg.Image)
+    cache = message_filters.Cache(sub, 100)
+
+In this example, the Cache stores the last 100 messages received on my_topic, and myCallback is called on the addition of every new message. The user can then make calls like cache.getInterval(start, end) to extract part of the cache.
+If the message type does not contain a header field that is normally used to determine its timestamp, and the Cache is contructed with allow_headerless=True, the current ROS 2 time is used as the timestamp of the message. This is currently only available in Python.
+
+6. PolicyBased Synchronizers
+----------------------------
+The Synchronizer filter synchronizes incoming channels by the timestamps contained in their headers, and outputs them in the form of a single callback that takes the same number of channels. The C++ implementation can synchronize up to 9 channels.
+
+The Synchronizer filter is templated on a policy that determines how to synchronize the channels. There are currently two policies: ExactTime and ApproximateTime.
```

Comment:
There is a third one [ApproximateEpsilonTime](https://github.com/ros2/message_filters/blob/rolling/include/message_filters/sync_policies/approximate_epsilon_time.h). Not need to write a independent tutorial in C++ and Python, you can open an issue with this task